compiler/rustc_arena/src/lib.rs

```diff
@@ -22,6 +22,7 @@
 #![feature(strict_provenance)]
 #![deny(rustc::untranslatable_diagnostic)]
 #![deny(rustc::diagnostic_outside_of_impl)]
+#![allow(clippy::mut_from_ref)] // Arena allocators are one of the places where this pattern is fine.
```

Comment:
Isn't this redundant with the other outer allows in this file?

Reply:
Lint levels aren't hygienic, so the macro sadly still needs them.

Reply:
Ew.